src/Air.zig

```diff
@@ -517,14 +518,6 @@ pub const Inst = struct {
         /// Read a value from a pointer.
         /// Uses the `ty_op` field.
         load,
-        /// Converts a pointer to its address. Result type is always `usize`.
-        /// Pointer type size may be any, including slice.
-        /// Uses the `un_op` field.
-        int_from_ptr,
-        /// Given a boolean, returns 0 or 1.
-        /// Result type is always `u1`.
-        /// Uses the `un_op` field.
-        int_from_bool,
```

Comment:
Is the idea here that handling pointer and bool along with other integer types in the same AIR instruction works better in your isel DSL?

Reply:
I was discussing this change with Jacob earlier -- it's more just weird that these had separate instructions in the first place. We use the `bitcast` instruction for plenty of other bit-reinterpretation operations which nonetheless have special names in Zig, like pointer casts. It's just more consistent to do it this way, and I think simpler.

Reply:
No, as mentioned in the commit message, these only exist (unlike e.g. `ptr_from_int`) because they have a fixed result type, which no longer saves any space in the Air.  And in order to extend these operations to vectors they need a variable result type, so I'm just using `bitcast` for all of these cases.

Reply:
I can explain the original motivation, which I think remains relevant to the rest of the pipeline:

Generally, it is better to branch once in the switch prong and have different implementations, rather than branching twice.

For example:

```zig
switch (tag) {
    .c => c(tag),
    // ...
}
fn c(tag) void {
    if (tag) {
        foo();
    } else {
        bar();
    }
}
```

:arrow_down: 

```zig
switch (tag) {
    .a => a(),
    .b => b(),
    // ...
}
fn a() void {
    foo();
}
fn b() void {
    bar();
}
```

If you look at, for instance, the llvm.zig changes, they fit this pattern in reverse. What was once one branch is now two branches.

To be clear, I think if the x86 isel dsl wants it this way that's a compelling reason to change it.

Reply:
> If you look at, for instance, the llvm.zig changes, they fit this pattern in reverse. What was once one branch is now two branches.

I mean, that seems like an LLVM-ism more than anything. ptr -> int is a trivial conversion for most backends; meanwhile, some conversions that *already* used `bitcast` *did* require branching in self-hosted backends, such as `f80` -> `u80` (example stolen from my earlier conversation with jacobly).

Reply:
I agree that AIR is currently overfit to the LLVM backend

Reply:
Frankly, I don't care that much about llvm weirdness, the vast majority of the backends are not going to care about `usize` &rightarrow; `*u8` vs `*u8` &rightarrow; `usize` vs `*u8` &rigtharrow; `*u32`.  Most of the special cases I had to add were simply backends that didn't already support `bitcast` with arbitrary types, and so I have to special case this one specific conversion to maintain feature parity.

Reply:
This is actually the same reason why I think `bool_and` and `bool_or` should be deleted.  Every single backend handles them identically to `bit_and` and `bit_or`, and the bitwise operations are already defined to return the same result type as the type of the lhs.

Reply:
ok sounds good then